src/modules/secrets/Elsa.Secrets/Providers/SecretsProvider.cs

```diff
@@ -40,16 +46,50 @@ public SecretsProvider(ISecretsManager secretsManager, IEnumerable<ISecretValueF
             return null;
         }
 
-        public async Task<ICollection<string>> GetSecrets(string type)
+        public async Task<ICollection<string>> GetSecretsAsync(string type)
+        {
+            var secrets = await _secretsManager.GetSecrets(type);
+
+            var formatter = _valueFormatters.FirstOrDefault(x => x.Type == type);
+
+            if (formatter != null)
+                return secrets.Select(x => formatter.FormatSecretValue(x).Result).ToArray();
+
+            return new List<string>();
+        }
+
+        public async Task<string?> GetSecretByNameAsync(string type, string name)
         {
             var secrets = await _secretsManager.GetSecrets(type);
-            return secrets.Select(x => x.Id).ToArray();
+
+            var formatter = _valueFormatters.FirstOrDefault(x => x.Type == type);
+
+            if (formatter != null)
+                return secrets.Where(x => x.Name?.Equals(name, StringComparison.InvariantCultureIgnoreCase) == true && x.Type?.Equals(type, StringComparison.InvariantCultureIgnoreCase) == true)
+                    .Select(x => formatter.FormatSecretValue(x).Result).FirstOrDefault();
+
+            return null;
         }
 
-        public async Task<ICollection<(string, string)>> GetSecretsForSelectListAsync(string type)
+        public async Task<IDictionary<string, string>> GetSecretsDictionaryAsync(string type)
         {
             var secrets = await _secretsManager.GetSecrets(type);
-            return secrets.Select(x => (x.Name ?? x.DisplayName, x.Name)).ToArray();
+
+            var formatter = _valueFormatters.FirstOrDefault(x => x.Type == type);
+
+            try
+            {
+                if (formatter != null)
+                    return secrets.ToDictionary(x => x.Name ?? x.DisplayName ?? x.Id, x => formatter.FormatSecretValue(x).Result);
```

Comment:
This actually breaks the referencing by name. It was intentionally changed compared to the original implementation. Formatting every secret for the options select list will just copy secret value during design time and will not be managable in future. In addition this complete breaks Oauth2 secret which only works when evaluated during runtime.

Reply:
This can be changed to the way you changed it to, value being the name of the secret. But, I would be sure to use the new format `<secret type>:<secret name>`. So that when the expression is evaluated, it can determine exactly what secret to use when there may be two secrets with the same name, but different type.

Reply:
@ke-real did you happen to update this, or should I do it real quick for 2.10 release?

Reply:
@johnwc it should be fixed in #3575 but we can also make a smaller PR if needed.